Don't send empty bodies

simmel opened the pull request:
When using this conf:

```
  email {
    […]
    htmlbody => "<h1>example</h1><b>html</b>"
  }
```

with current code you'd also attach an empty text/plain mail.

With this patch only the attachment, text/plain or text/html, which is
specified via `body => ""` or `htmlbody => ""` is sent.

purbon commented:
Thanks a lot for your contribution @simmel , in order to move forward with your PR is going to be necessary to sign the CLA agreement, you can find more information at https://www.elastic.co/contributor-agreement.

On the other side, it would be super nice if you can add test for this change. We try to enforce that all PR introduce some kind of testing, so we're able to increase the overall quality. Don't hesitate to ask any question regarding you might have, more than looking forward to help.

/cheers

simmel commented:
*sigh* I commited it with my personal email again, didn't I? = )
I'll fix that on monday.

Since the change doesn't add anything, just changes the default I
thought the already existing tests were sufficient? All tests pass.

simmel commented:
Actually, I did commit it with my job email.

We have signed an CLA, but one for our company which I was included in.
I can't find any "id" for the CLA but the company is "Stockholm
University" and it was signed 2015-02-17.

Did we do something wrong when I was attached to the "Schedule A"-part
of the SLAs?

simmel commented:
Oh wait, the transaction ID is XJZHHY4333R222U.

purbon commented:
@simmel ok, let me see then :+1: thanks for your feedback.

elasticsearch-release commented:
Jenkins standing by to test this. If you aren't a maintainer, you can ignore this comment. Someone with commit access, please review this and clear it for Jenkins to run; then  say 'jenkins, test it'.

simmel commented:
Any news @purbon ?

purbon commented:
please jenkins, test this.

simmel commented:
Did that trigger Jenkins? @elasticsearch-release said 'jenkins, test it' but ¯\_(ツ)_/¯

Also, if the missing CLA is fixed maybe the label can be removed?
And since this doesn't add any new functionality no new tests are needed either so that label can be removed too.

Thanks!

simmel commented:
I'll fix the conflicts @jsvd

simmel commented:
@jsvd And CLA can't be missing, we have signed it and I have commited with the correct mailaddress.

jsvd commented:
@simmel I think the issue we're seeing here is that the email from the commit isn't the one associated with your github account

simmel commented:
Like I said, I have commited with the correct emailaddress:
```
$ git clone git@github.com:simmel/logstash-output-email.git
Cloning into 'logstash-output-email'...
remote: Counting objects: 384, done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 384 (delta 4), reused 17 (delta 4), pack-reused 364
Receiving objects: 100% (384/384), 66.75 KiB | 564.00 KiB/s, done.
Resolving deltas: 100% (165/165), done.
$ cd logstash-output-email
$ git co dont_send_empty_bodies
$ git log -n 1 --pretty=short
commit bc35caa5a62d1b4ead11dd85ce9d2c9962ef2eb6 (HEAD -> dont_send_empty_bodies, origin/dont_send_empty_bodies)
Author: Simon Lundström <simlu@su.se>

    Don't send empty bodies
$
```

The Transaction ID is CBJCHBCAABAArwhazwMmCOuIx1PFdJhx58iBbf9c1H65.

simmel commented:
> associated with your github account

Yes, it's added at https://github.com/settings/emails *but* it's *not* my primary email address. Are you looping over all mail addresses?

jsvd commented:
@simmel I spoke with the person running clachecker and the company wide agreement still requires individual signatures under the company's one, and there's a special option for that:

<img width="728" alt="screen shot 2018-01-11 at 13 33 42" src="https://user-images.githubusercontent.com/31809/34828891-ea79682c-f6d6-11e7-856c-6de8f57c9acc.png">

simmel commented:
Aaaah. Så just because I'm listed on the last page of the first signing 
doesn't mean that I've signed at all?

On Thu, 2018-01-11 at 05:54:37 -0800, João Duarte wrote:
> @simmel I spoke with the person running clachecker and the company wide agreement still requires individual signatures under the company's one, and there's a special option for that:
> 
> <img width="728" alt="screen shot 2018-01-11 at 13 33 42" src="https://user-images.githubusercontent.com/31809/34828891-ea79682c-f6d6-11e7-856c-6de8f57c9acc.png">
> 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/logstash-plugins/logstash-output-email/pull/27#issuecomment-356940287

simmel commented:
I just figured that it was when you wanted to add additional signers that you didn't upfront.

simmel commented:
Why is the first page there then?

simmel commented:
I've signed it now.

Since there has been some changes/additions in the code which further complicated the logic I'm gonna rewrite my patch and add tests tomorrow.

jsvd commented:
@simmel Thank you for your patience

simmel commented on spec/outputs/email_spec.rb:
```diff
@@ -101,8 +101,61 @@
 
         expect(message_observer.messages.size).to eq(1)
         expect(message_observer.messages[0].subject).to eq("Hello World")
-        expect(message_observer.messages[0].body.decoded).to eq(craft_multi_part_email('', '<h1>hello</h1>', message_observer.messages[0].content_type))
```
I had to change this test because it was wrong. It send in the `text/plain` content-type via `message_observer.messages[0].content_type` since before this patch we sent an empty `text/plain` mail even though we only wanted to send an `text/html` mail.

simmel commented:
I'm not sure why Github still thinks that there's a conflict since I've rebased master.

Also, I'm sorry for removing trailing spaces in this PR and not a seperate but after I rewrote the logic and the tests there were only two left so I removed them = /

simmel commented:
Any news @jsvd?

simmel commented:
Any news?


On Thu, 11 Jan 2018, at 15:43, João Duarte wrote:
> @simmel[1] Thank you for your patience


> — You are receiving this because you were mentioned. Reply to this
> email directly, view it on GitHub[2], or mute the thread[3].> 


Links:

  1. https://github.com/simmel
  2. https://github.com/logstash-plugins/logstash-output-email/pull/27#issuecomment-356954311
  3. https://github.com/notifications/unsubscribe-auth/AALqfHjsazmoRV0UZQJO4GF18HdWIMZeks5tJh4jgaJpZM4Fv1wJ
